### DISCUSSION TITLE
It is possible to read error integral from iCub motor control boards?

### DISCUSSION THREAD

**Discussion by traversaro** [robotology/community#494] [Q&A] [answered]
When a iCub joint is controlled in IPositionDirect control, I guess that internally one control board is implementing an PID controller, and as part of this controller the board is storing an **integral of the position error**. 

I wonder if there is some way (even extremely low frequency) to access this  **integral of the position error** from user code that is communicating with the robot via YARP?  Thanks in advance!

**Comment by pattacini** [comment, marked as answer]
I don't think so.

The motor interface closest to that goal would be [`IPidControl`][1].
Anyway, I don't see any method that could do the job.

That said, it might be interesting to have this functionality available.

cc @randaz81 

[1]: http://yarp.it/git-master/classyarp_1_1dev_1_1IPidControl.html
